include/hyprutils/os/FileDescriptor.hpp

```diff
@@ -1,6 +1,11 @@
 #pragma once
 
 #include <fcntl.h>
+
+#ifndef F_DUPFD_CLOEXEC
```

Comment:
I don't think we should define this at a library level. This should be only for .cpp files, we don't want to propagate this to users